### PR TITLE
Try to build Mono module

### PR DESCRIPTION
**Pull request by ricvelozo**
I managed to fix `monosgen-2` error (#8). Now I'm stuck on the following NuGet error:

```console
scons: *** [bin/GodotSharp/Tools/GodotSharpTools.dll] /usr/lib/sdk/mono5/lib/nuget/nuget.exe: Exec format error
```

Also, I temporarily changed the runtime version to be compatible with `org.freedesktop.Sdk.Extension.mono5` ([#7](https://github.com/flathub/org.freedesktop.Sdk.Extension.mono5/issues/7)).

### CODE REVIEW

**Comment by ricvelozo** [comment]
Godot 3.2 ([currently in beta](https://godotengine.org/article/dev-snapshot-godot-3-2-beta-4)) will need Mono 6 runtime because of WebAssembly support.

**Comment by ricvelozo** [comment]
Compilation errors have been fixed.

Now, when I run `bin/godot.x11.tools.* --generate-mono-glue modules/mono/glue`, I get:

```
Godot Engine v3.2.1.stable.mono.flathub - https://godotengine.org
ERROR: initialize: X11 Display is not available
   At: platform/x11/os_x11.cpp:132.
```

**Comment by ricvelozo** [comment]
I'd added build permissions to enable use OpenGL, to generate the "Mono glue". On my machine, I got that error:

```
Godot Engine v3.2.1.stable.mono.flathub - https://godotengine.org
ERROR: initialize: Condition "!fbc" is true. Returned: ERR_UNCONFIGURED
   At: platform/x11/context_gl_x11.cpp:158.
ERROR: is_viable: Error initializing GLAD
   At: drivers/gles3/rasterizer_gles3.cpp:141.
```

But, on Flathub I couldn't even compile the first module:

```
flatpak-builder: Building glu
FB: Running: tar xf ...
flatpak-builder: Cleanup glu
Error: Can't specify build-args in sandboxed build
```

**Comment by ricvelozo** [comment]
On my machine:

Captura de tela de 2020-05-17 23-47-55

**Comment by Calinou** [comment]
If generating Mono glue proves to be too much of an issue, we could build it on a local machine and commit the generated glue to this repository.

File: org.godotengine.Godot.yaml [hunk 1]
```diff
@@ -3,9 +3,19 @@ runtime: org.freedesktop.Platform
...
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.mono5
 command: godot
 
 build-options:
+  append-path: /usr/lib/sdk/mono5/bin
+  append-ld-library-path: /usr/lib/sdk/mono5/lib
+  build-args:
+    - --share=ipc
+    - --socket=x11
```
Comment (TingPing) [inline org.godotengine.Godot.yaml:17]: No idea if OpenGL can work but try running under `xvfb-run`?

Comment (ricvelozo) [reply org.godotengine.Godot.yaml]: I tried, but the `xvfb-run` is not in the runtime.

Comment (barthalion) [reply org.godotengine.Godot.yaml]: You can bundle it only for the build. `cleanup: ["*"]` will delete it afterwards.

Comment (ricvelozo) [reply org.godotengine.Godot.yaml]: Yes, I will try today after sleeping haha.

Comment (ricvelozo) [reply org.godotengine.Godot.yaml]: Using `Xvfb`, I got that error:

```
Godot Engine v3.2.1.stable.mono.flathub - https://godotengine.org
ERROR: initialize: X11 Display is not available
   At: platform/x11/os_x11.cpp:132.
```

Comment (PureTryOut) [reply org.godotengine.Godot.yaml]: You can generate the intermediate binaries with platform=server, then you don't need X11 to generate te glue. Just make sure you generate the final binaries with platform=x11 again.

Comment (ricvelozo) [reply org.godotengine.Godot.yaml]: I sent a new PR but it didn't trigger a build. :confused:

Comment (ricvelozo) [reply org.godotengine.Godot.yaml]: The error now: dotnet/msbuild#2423 / godotengine/godot#40204.

**Comment by Calinou** [comment]
Superseded by https://github.com/flathub/org.godotengine.Godot/pull/109. Thanks for the contribution nonetheless!

**Pull request closed without merging**